pkg/services/rocketchat.go

```diff
@@ -0,0 +1,134 @@
+package services
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+	texttemplate "text/template"
+
+	"github.com/RocketChat/Rocket.Chat.Go.SDK/models"
+	"github.com/RocketChat/Rocket.Chat.Go.SDK/rest"
+	log "github.com/sirupsen/logrus"
+)
+
+type RocketChatNotification struct {
+	Attachments string `json:"attachments,omitempty"`
+}
+
+func (n *RocketChatNotification) GetTemplater(name string, f texttemplate.FuncMap) (Templater, error) {
+	rocketChatAttachments, err := texttemplate.New(name).Funcs(f).Parse(n.Attachments)
+	if err != nil {
+		return nil, err
+	}
+	return func(notification *Notification, vars map[string]interface{}) error {
+		if notification.RocketChat == nil {
+			notification.RocketChat = &RocketChatNotification{}
+		}
+		var rocketChatAttachmentsData bytes.Buffer
+		if err := rocketChatAttachments.Execute(&rocketChatAttachmentsData, vars); err != nil {
+			return err
+		}
+
+		notification.RocketChat.Attachments = rocketChatAttachmentsData.String()
+
+		return nil
+	}, nil
+}
+
+type RocketChatOptions struct {
+	Alias              string   `json:"alias"`
+	Email              string   `json:"email"`
+	Password           string   `json:"password"`
+	Icon               string   `json:"icon"`
+	Avatar             string   `json:"avatar"`
+	Token              string   `json:"token"`
+	ServerUrl          string   `json:"serverUrl"`
+	SigningSecret      string   `json:"signingSecret"`
+	Channels           []string `json:"channels"`
+	InsecureSkipVerify bool     `json:"insecureSkipVerify"`
```

Comment:
I missed it, this field is not using.
Would you use like below, please?

https://github.com/argoproj/notifications-engine/blob/c1ebfd813101de8a250e35296c5b1bde0fcd4a72/pkg/services/slack.go#L81-L85

Reply:
Hm... I'm not sure that RocketChat client support it i will look

Reply:
As i see RocketChat client has no support on such an option. So I wanna remove it

Reply:
I thinks there is no problem because the user can connect using http.
https://github.com/RocketChat/Rocket.Chat.Go.SDK/blob/10ab4d695d6089499cd2c9aab9ff00fe05952cd8/rest/client.go#L75-L89